juju/client/connection.py

```diff
@@ -324,70 +364,50 @@ def https_connection(self):
             "/model/{}".format(self.uuid)
             if self.uuid else ""
         )
-        return conn, self.http_headers(), path
+        return conn, self._http_headers(), path
 
     async def clone(self):
         """Return a new Connection, connected to the same websocket endpoint
         as this one.
 
         """
-        return await Connection.connect(
-            self.endpoint,
-            self.uuid,
-            self.username,
-            self.password,
-            self.cacert,
-            self.macaroons,
-            self.loop,
-            self.max_frame_size,
+        endpoint, kwargs = self.connect_params()
+
+        return await Connection.connect(endpoint, **kwargs)
+
+    def connect_params(self):
```

Comment:
We also use it to call other methods such as Model._connect_direct, which wouldn't work with that change.
It can also be useful for debugging - I think it's probably reasonable as is.